src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -302,7 +302,7 @@
 ;; Functions to ensure that the catalog structure is coherent.
 
 (def ^:const tag-pattern
-  #"\A[a-z0-9_][a-z0-9_:\-.]*\Z")
+  #"\A[[\p{L}&&[^\p{Lu}]]0-9_][[\p{L}&&[^\p{Lu}]]0-9_:\-.]*\Z")
```

Comment:
What's the matching, presumably ruby pattern?  And might be good to include a comment or docstring describing what this has to match on the puppet side (better yet if there's a solid doc link).

Reply:
Oh, and didn't mean that we should include the current ruby pattern here, just wanted to look at it myself.

Reply:
Current ruby pattern is below (taken from here https://github.com/puppetlabs/puppet/blob/2bec43a16c472f0ef1a6f9b79f7eff929f6e95dd/lib/puppet/util/tagging.rb#L4)

`/\A[[:alnum:]_][[:alnum:]_:.-]*\Z/u`

Reply:
I need to change the way we match digits, Puppet now allows unicode digits as well as `0-9`